subsys/bluetooth/host/hci_core.c

```diff
@@ -695,6 +690,32 @@ static bool bt_le_adv_random_addr_check(const struct bt_le_adv_param *param)
 		    param->id != BT_ID_DEFAULT) {
 			return false;
 		}
+	} else if (IS_ENABLED(CONFIG_BT_SCAN_WITH_IDENTITY) &&
+		   atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING) &&
+		   bt_dev.id_addr[BT_ID_DEFAULT].type == BT_ADDR_LE_RANDOM) {
+		/* Scanning with random static identity. Stop the advertiser
+		 * from overwriting the passive scanner identity address.
+		 * In this case the LE Set Random Address command does not
+		 * protect us in the case of a passive scanner.
+		 * Explicitly stop it here.
+		 */
+
+		if (!(param->options & BT_LE_ADV_OPT_CONNECTABLE) &&
+		    (param->options & BT_LE_ADV_OPT_USE_IDENTITY)) {
+			/* Attempt to set non-connectable NRPA */
+			return false;
+		}
+
+
+		if (((param->options & BT_LE_ADV_OPT_CONNECTABLE) ||
```

Comment:
`((param->options & BT_LE_ADV_OPT_CONNECTABLE) || !(param->options & BT_LE_ADV_OPT_USE_IDENTITY))`

Is redundant as it evaluates to NOT operation of line 703, instead use  `else if` clause.

Reply:
Thanks. Updated.